Cython/Includes/numpy/__init__.pxd

```diff
@@ -244,14 +244,27 @@ cdef extern from "numpy/arrayobject.h":
 
         cdef:
             # Only taking a few of the most commonly used and stable fields.
-            # One should use PyArray_* macros instead to access the C fields.
             char *data
-            int ndim "nd"
-            npy_intp *shape "dimensions"
-            npy_intp *strides
             dtype descr  # deprecated since NumPy 1.7 !
             PyObject* base
 
+            @property
+            cdef int ndim(self):
```

Comment:
Just wondering, but do you think it would help to declare these `inline`?

Reply:
maybe. I would have to look at the C-code from a real usage to see how much overhead would be removed